lib/reek/configuration/configuration_file_finder.rb

```diff
@@ -14,41 +14,25 @@ module Configuration
     # The order in which ConfigurationFileFinder tries to find such a
     # configuration file is exactly like above.
     module ConfigurationFileFinder
-      class << self
-        def find(options)
-          configuration_by_cli(options) ||
-            configuration_in_file_system    ||
-            configuration_in_home_directory
-        end
-
-        private
-
-        def configuration_by_cli(options)
-          return unless options # Return gracefully allowing calls without app context
-          config_file_option = options.config_file
-          return unless config_file_option
-          path_name = Pathname.new config_file_option
-          raise ArgumentError, "Config file #{path_name} doesn't exist" unless path_name.exist?
-          path_name
-        end
-
-        def configuration_in_file_system
-          detect_or_traverse_up Pathname.pwd
-        end
+      module_function
 
-        def configuration_in_home_directory
-          detect_configuration_in_directory Pathname.new(Dir.home)
-        end
+      # FIXME: switch to kwargs on upgrade to Ruby 2 and drop `dirs.fetch` calls:
+      # def find(options, current: Pathname.pwd, home: Pathname.new(Dir.home))
+      def find(options, dirs = {})
+        current = dirs.fetch(:current) { Pathname.pwd           }
+        home    = dirs.fetch(:home)    { Pathname.new(Dir.home) }
+        find_by_cli(options) || find_by_dir(current) || find_by_dir(home)
+      end
 
-        def detect_or_traverse_up(directory)
-          file = detect_configuration_in_directory(directory)
-          return file unless file.nil?
-          return if directory.root?
-          detect_or_traverse_up directory.parent
-        end
+      def find_by_cli(options)
+        options && options.config_file
+      end
 
-        def detect_configuration_in_directory(directory)
-          directory.children.select(&:file?).find { |path| path.to_s.end_with?('.reek') }
+      def find_by_dir(start)
+        start.ascend do |dir|
```

Comment:
Me neither – half of this PR was reading pathname’s docs. :)

Fun fact: `Pathname#ascend` is so old it requires a block; if it returned an `Enumerator` we could do this instead:

```ruby
      def find_by_dir(start)
        start.ascend.find do |dir|
          dir.children.select(&:file?).find { |file| file.to_s.end_with?('.reek') }
        end
      end
```